apis/Google.Cloud.AspNetCore.DataProtection.SecretManager/Google.Cloud.AspNetCore.DataProtection.SecretManager/GoogleCloudDataProtectionBuilderExtensions.cs

```diff
@@ -0,0 +1,78 @@
+/// </summary>
+public static class GoogleCloudDataProtectionBuilderExtensions
+{
+    /// <summary>
+    /// Configures data protection system to persist keys in Google Cloud Secret Manager.
+    /// </summary>
+    /// <param name="builder">The data protection builder to configure. Must not be null.</param>
+    /// <param name="projectId">The project id in which the keys are stored. Must not be null</param>
+    /// <param name="secretName">The name of the secret in which to store the keys. Must not be null.</param>
+    public static IDataProtectionBuilder PersistKeysToGoogleCloudSecretManager(
+            this IDataProtectionBuilder builder, string projectId, string secretName) =>
+            PersistKeysToGoogleCloudSecretManager(builder, projectId, secretName, null);
+
+    /// <summary>
+    /// Configures data protection builder <see cref="IDataProtectionBuilder"/> to persist keys in Google Cloud Secret Manager.
+    /// </summary>
+    /// <remarks>
+    /// <para>
+    /// If <paramref name="client"/> is null, the client is constructed as follows:
+    /// <list type="bullet">
+    ///   <item><description>If a <c>SecretManagerServiceClient</c> is configured via dependency injection, that is used.</description></item>
+    ///   <item><description>If a <c>GoogleCredential</c> is configured via dependency injection, that is used to construct a <c>SecretManagerServiceClient</c>.</description></item>
+    ///   <item><description>Otherwise, the default credentials are used to construct a <c>SecretManagerServiceClient</c>.</description></item>
+    /// </list>
+    /// </para>
+    /// </remarks>
+    /// <param name="builder">The data protection builder to configure. Must not be null.</param>
+    /// <param name="projectId">The project id in which the keys are stored. Must not be null</param>
+    /// <param name="secretName">The name of the secret in which to store the keys. Must not be null.</param>
+    /// <param name="client">The Google Cloud Secret Manager client to use for requests. May be null in which case the client will
+    /// be fetched from dependency injection or created with the default credentials.</param>
+    /// <returns>The same builder, for chaining purposes</returns>
+    public static IDataProtectionBuilder PersistKeysToGoogleCloudSecretManager(
+            this IDataProtectionBuilder builder, string projectId, string secretName, SecretManagerServiceClient client)
+    {
+        builder.Services.AddSingleton<IConfigureOptions<KeyManagementOptions>>(services =>
+        {
+            if (client == null)
+            {
+                client = services.GetService<SecretManagerServiceClient>();
+                if (client == null)
+                {
+                    var credential = services.GetService<GoogleCredential>();
+                    // If credential is null, this will use the default credentials automatically.
+                    client = SecretManagerServiceClient.Create();
```

Comment:
You're not using credential here at all.

Reply:
rectified